src/aiori-DAOS.c

```diff
@@ -334,38 +362,38 @@ DAOS_Fini()
 static void
 gen_oid(const char *name, daos_obj_id_t *oid)
 {
-
 	oid->lo = d_hash_murmur64(name, strlen(name), IOR_DAOS_MUR_SEED);
 	oid->hi = 0;
 
 	daos_array_generate_id(oid, objectClass, true, 0);
 }
 
-static void *
-DAOS_Create(char *testFileName, IOR_param_t *param)
+static aiori_fd_t *
+DAOS_Create(char *testFileName, int flags, aiori_mod_opt_t *param)
 {
+        DAOS_options_t *o = (DAOS_options_t*) param;
 	daos_obj_id_t	oid;
 	int		rc;
 
 	/** Convert file name into object ID */
 	gen_oid(testFileName, &oid);
-
+               
 	/** Create the array */
-	if (param->filePerProc || rank == 0) {
-		rc = daos_array_create(coh, oid, DAOS_TX_NONE, 1, o.chunk_size,
+	if (hints->filePerProc || rank == 0) {
+		rc = daos_array_create(coh, oid, DAOS_TX_NONE, 1, o->chunk_size,
 				       &aoh, NULL);
 		DCHECK(rc, "Failed to create array object\n");
 	}
 
 	/** Distribute the array handle if not FPP */
-	if (!param->filePerProc)
+	if (!hints->filePerProc)
 		HandleDistribute(&aoh, ARRAY_HANDLE);
 
-	return &aoh;
+	return (aiori_fd_t*)(&aoh);
```

Comment:
Here is the line that fits together with my concern.

Reply:
yes, you are referring to the DAOS array driver.
This is a legacy implementation that was done using the DAOS Array API (before the DFS layer was implemented). As I mentioned, the DFS driver is the way to go, and i will soon push a patch to remove this entire driver.